src/course/course.controller.ts

```diff
@@ -0,0 +1,44 @@
+import { Controller, Get, Post, Body, Param, Put, Delete } from '@nestjs/common';
+import { CourseService } from './course.service';
+import { Course } from './entities/course.entity';
+
+@Controller('courses')
+export class CourseController {
+  constructor(private courseService: CourseService) {}
+
+  @Post()
+  async create(@Body() course: Course): Promise<Course> {
+    return this.courseService.create(course);
+  }
+
+  @Get()
+  async findAll(): Promise<Course[]> {
+    return this.courseService.findAll();
+  }
+
+  @Get('id/:id')
+  async findById(@Param('id') id: number): Promise<Course> {
+    return this.courseService.findById(id);
+  }
+
+  @Get('teacher/:teacherName')
+  async findByTeacher(@Param('teacherName') teacherName: string): Promise<Course> {
```

Comment:
`findByTeacher`'s return type should be `Promise<Course[]>` to indicate that there is a "one to many" relation between teacher and courses